### PR TITLE
Add test vectors for ChaCha20 counter overflow

### PR DESCRIPTION
**Pull request by facutuesca**
Since we are adding support for ChaCha20 with different backends (https://github.com/pyca/cryptography/pull/9209), it's important we make sure that the behavior across them is the same.

Particularly with ChaCha20, where different implementations use different counter sizes, and therefore their behavior differs during counter overflow (see https://github.com/pyca/cryptography/issues/8956 for more context).

This PR adds test vectors that force a counter overflow (both for 64-bit and 32-bit counters). It also adapts the existing test vectors to match the actual implementation (using a 64/64 bit counter/nonce split, rather than a 32/96 split)

### CODE REVIEW

File: vectors/cryptography_vectors/ciphers/ChaCha20/counter-overflow.txt
[the remark this answers is not in the document]
Comment (alex) [reply vectors/cryptography_vectors/ciphers/ChaCha20/counter-overflow.txt:1]: Yes

[the remark this answers is not in the document]
Comment (facutuesca) [reply vectors/cryptography_vectors/ciphers/ChaCha20/counter-overflow.txt:1]: Done! I also added scripts to generate and verify the vectors

**Review by woodruffw — APPROVED** [review]
LGTM, needs ultimate approval from @reaperhulk and @alex 🙂

**Comment by reaperhulk** [comment]
Go uses the 64-bit counter right? I'd like to see a verifier from another implementation here. Although I don't see any errors in the Python code, demonstration of it behaving as expected on a known 64-bit counter implementation is a more powerful demo.

**Review by reaperhulk — APPROVED** [review]

**Pull request merged**